### ISSUE TITLE
Middleware data never sent on Android

### ISSUE THREAD

**Issue by jpolo** [segmentio/analytics-react-native#48]
Example
```javascript
import SegmentAnalytics from '@segment/analytics-react-native';

SegmentAnalytics.middleware(({ next, context }) => next({
    ...context,
    anyData: true,
}));

SegmentAnalytics.track(...) // -> context.anyData is sent on iOS but not on Android
```

**Comment by sampsonjoliver** [comment]
Same issue here. I got around this by adding the additional properties to `data` with the following, but it's not ideal:

```
analytics.middleware((payload) => {
  const newProps = { ... };
  const { next, context } = payload;

  if (payload.type === 'track') {
    return next(context, {
      event: payload.data.event,
      properties: {
        ...payload.data.properties,
        ...newProps,
      },
    });
  } else if (payload.type === 'screen') {
    return next(context, {
      name: payload.data.name,
      properties: {
        ...payload.data.properties,
        ...newProps,
      },
    });
  } else {
    return next(context);
  }
});
```

**Comment by sampsonjoliver** [comment]
It's [quite plain to see](https://github.com/segmentio/analytics-react-native/blob/master/packages/core/android/src/main/java/com/segment/analytics/reactnative/core/RNAnalyticsModule.kt#L98) that the Android native code doesn't actually pass the context data to its native calls to the tracker.

Presumably this is because the native Android client only takes `properties` and no `context` in its track calls ([seen here](https://github.com/segmentio/analytics-android/blob/master/analytics/src/main/java/com/segment/analytics/Analytics.java#L608)), and instead relies on a `Analytics.with(context).track(event, properties)` syntax. However the `context` there is expecting the Android app context.

Instead it looks to me the best approach within this library would be to use the `options` arg in the android native client, as [per the documentation ](https://github.com/segmentio/analytics-android/blob/master/analytics/src/main/java/com/segment/analytics/Options.java#L127) it allows adding values to just this one tracking call

i.e. (and I apologise, my Kotlin is a bit rusty)
```
    @ReactMethod
    fun track(event: String, properties: ReadableMap, context: ReadableMap) = 
        analytics.track(event, Properties() from properties, Options().apply { opts ->
            context.forEach { opts.putContext(it.key, it.value) }
    })
```

**Comment by ericketts** [comment]
huh yeah that PR you linked @sampsonjoliver was blocked by another PR that.. has been merged... so its puzzling why the fix PR that was blocked hasn't now been merged... puzzling indeed...

**Comment by sampsonjoliver** [comment]
Yeah @ericketts I'm not sure why @fathyb closed that one, but it's good to know I'm not the only one puzzled 🤷‍♂  

Either way, here's a fresh one that's pretty much the same thing again! https://github.com/segmentio/analytics-react-native/pull/66
